src/foam/nanos/notification/email/DAOResourceLoader.java

```diff
@@ -0,0 +1,42 @@
+package foam.nanos.notification.email;
+
+import com.google.common.base.Optional;
+import foam.dao.DAO;
+import org.jtwig.resource.loader.ResourceLoader;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+public class DAOResourceLoader
+    implements ResourceLoader
+{
+  protected DAO dao_;
+
+  public DAOResourceLoader(DAO dao) {
+    this.dao_ = dao;
+  }
+
+  @Override
+  public Optional<Charset> getCharset(String s) {
+    return Optional.absent();
+  }
+
+  @Override
+  public InputStream load(String s) {
+    EmailTemplate template = (EmailTemplate) dao_.find(s);
+    return new ByteArrayInputStream(template.getBody().getBytes(StandardCharsets.UTF_8));
+  }
+
+  @Override
+  public boolean exists(String s) {
+    return ( dao_.find(s) != null );
```

Comment:
@kgrgreer Should I be caching the templates into a Map in this class?

Reply:
If StringBufferInputStream works, then the overhead is negligible, so I wouldn't bother.